Marlin/src/module/temperature.cpp

```diff
@@ -2246,65 +2316,83 @@ void Temperature::disable_all_heaters() {
     if (PENDING(ms, next_max6675_ms[hindex])) return int(MAX6675_TEMP(hindex));
     next_max6675_ms[hindex] = ms + MAX6675_HEAT_INTERVAL;
 
-    #if HAS_MAX31865
-      Adafruit_MAX31865 &maxref = MAX6675_SEL(max31865_0, max31865_1);
-      const uint16_t max31865_ohms = (uint32_t(maxref.readRTD()) * MAX6675_SEL(MAX31865_CALIBRATION_OHMS_0, MAX31865_CALIBRATION_OHMS_1)) >> 16;
-    #endif
-
     //
     // TODO: spiBegin, spiRec and spiInit doesn't work when soft spi is used.
     //
-    #if !MAX6675_SEPARATE_SPI
+    #if !MAX6675_SEPARATE_SPI && !HAS_MAX31865 && !HAS_MAX31855 && !HAS_MAX6675_TEMP
       spiBegin();
       spiInit(MAX6675_SPEED_BITS);
     #endif
 
-    MAX6675_WRITE(LOW);  // enable TT_MAX6675
-    DELAY_NS(100);       // Ensure 100ns delay
+    #if !HAS_MAX31865 && !HAS_MAX31855 && !HAS_MAX6675_TEMP
+      MAX6675_WRITE(LOW);  // enable TT_MAX6675 
+      DELAY_NS(100);       // Ensure 100ns delay  
+    #endif
 
     // Read a big-endian temperature value
     max6675_temp = 0;
-    for (uint8_t i = sizeof(max6675_temp); i--;) {
-      max6675_temp |= TERN(MAX6675_SEPARATE_SPI, max6675_spi.receive(), spiRec());
-      if (i > 0) max6675_temp <<= 8; // shift left if not the last byte
-    }
+    #if !HAS_MAX31865 && !HAS_MAX31855 && !HAS_MAX6675_TEMP
```

Comment:
should not the following code be `#if / #elsif / #else`?

Reply:
NO, HAS_MAX??? only get set to 1 when it is active.  So Only one of those will be active at a time

Reply:
That #if statement is there to STOP the manipulation of SS_PIN when it is not required to get the communication done.  It take out code that is not needed to get perform the SPI communication.

Reply:
After looking the code over, I decided the line here in temeperature.cpp you are mentioning will never get executed.  So at the  moment it is commented out.  I left it in place, just in case I have over looked something.  I will delete the code that is commented out in temperature.cpp after I finish running this PR code on different MCU boards that I have in house.

So this item has been fixed.